uipyqt.py

```diff
@@ -0,0 +1,288 @@
+        self.group_box = QGroupBox(tactics.table_frame_name)
+        self.vbox = QHBoxLayout()
+
+        self.group_box.setLayout(self.vbox)
+        self.main_layout.addWidget(self.group_box)
+
+        # draw my hand
+        self.main_layout.addWidget(QLabel(tactics.my_hand_frame_name))
+        self.my_hand_layout=QHBoxLayout()
+
+
+        self.main_layout.addLayout(self.my_hand_layout)
+
+        # add action buttons
+        self.main_layout.addWidget(QLabel(tactics.actions_frame_name))
+
+        self.my_action_layout=QHBoxLayout()
+
+        # then add two buttons - for claiming the cards from the table
+        self.claim_cards_button = QPushButton(tactics.claim_cards_button_name)
+        self.claim_cards_button.clicked.connect(self.pressed_claim_cards)
+        self.my_action_layout.addWidget(self.claim_cards_button)
+        # ...and for laying the card on the table
+        self.lay_card_button = QPushButton(tactics.lay_card_button_name)
+        self.lay_card_button.clicked.connect(self.pressed_lay_card)
+        self.my_action_layout.addWidget(self.lay_card_button)
+        # and OK button
+        self.OK_button = QPushButton("OK")
+        self.OK_button.clicked.connect(self.pressed_OK)
+        self.my_action_layout.addWidget(self.OK_button)
+
+        self.main_layout.addLayout(self.my_action_layout)
+
+        self.setLayout(self.main_layout)
+        self.setWindowTitle("Scopa")
+
+    # removes all widgets from given layout
+    def remove_all_widgets(self, layout):
+        widget_range=reversed(range(layout.count()))
+        for i in widget_range:
+            widgetToRemove = layout.itemAt(i).widget()
+            button_name = widgetToRemove.text()
+            # remove it from the layout list
+            layout.removeWidget(widgetToRemove)
+            # remove it from the gui
+            widgetToRemove.setParent(None)
+
+    # this function clears the form from all items in the view
+    # empties all the global collections
+    # and sets all the flags to False
+    def clear_form(self):
+        self.remove_all_widgets(self.opponent_hand_layout)
+        self.opponent_buttons=[]
+
+        self.remove_all_widgets(self.vbox)
+        self.table_buttons=[]
+
+        self.remove_all_widgets(self.my_hand_layout)
+        self.my_hand_buttons=[]
+
+    # sets buttons "Claim cards" and "Lay card" as visible, OK as invisible
+    def enable_button_for_my_move(self):
+        self.claim_cards_button.setVisible(True)
+        self.lay_card_button.setVisible(True)
+        self.OK_button.setVisible(False)
+
+    # sets button OK as visible, "Claim cards" and "Lay card" as invisible
+    def enable_button_for_opponent_move(self):
+        self.claim_cards_button.setVisible(False)
+        self.lay_card_button.setVisible(False)
+        self.OK_button.setVisible(True)
+
+    # draw opponent hand making only the button indicated enabled and checked
+    def draw_opponent_hand(self, card_to_display = ""):
```

Comment:
Change naming from "draw" to "render" for this and subsequent methods in this file to avoid confusion with method names in scopa.py.